include/swift/Parse/Lexer.h

```diff
@@ -232,14 +233,13 @@ class Lexer {
 
   /// \brief Restore the lexer state to a given one, that can be located either
   /// before or after the current position.
-  void restoreState(State S) {
+  void restoreState(State S, bool emitDiagnostics = false) {
```

Comment:
`emitDiagnostics` makes me feel it always emit some diagnostics.
How about `enableDiagnostics`, or `suppressDiagnostics` (with inverted meaning)?